### PR TITLE
Produce ihex files usable with TI's Flash Programmer

### LINKED ISSUE #1058: Cortex M3 .hex files
The CC2538 and CC26xx ports are identical in terms of how they produce the final .bin and .hex

```
arm-none-eabi-objcopy -O ihex cc2538-demo.elf cc2538-demo.hex
arm-none-eabi-objcopy -O binary --gap-fill 0xff cc2538-demo.elf cc2538-demo.bin
```

In a nutshell, we create a .elf and from it we create a .bin and a .hex

The `.bin` works perfectly with CC2538 as well as with CC26xx. The .hex works fine with CC2538, but it fails miserably with CC26xx. TI's flash prog complains about an unknown record type, but I suspect this is not the primary cause.

I ran objdump on both hex files

```
$ arm-none-eabi-objdump -h cc26xx-demo.hex 

cc26xx-demo.hex:     file format ihex

Sections:
Idx Name          Size      VMA               LMA               File off  Algn
  0 .sec1         0000d6bd  00000000  00000000  00000000  2**0
                  CONTENTS, ALLOC, LOAD
  1 .sec2         00000008  0000d6c0  0000d6c0  00025bf6  2**0
                  CONTENTS, ALLOC, LOAD
  2 .sec3         00000054  0001ffac  0001ffac  00025c24  2**0
                  CONTENTS, ALLOC, LOAD
```

and this looks normal to me.

Any ideas?

### PR DESCRIPTION
**Pull request by g-oikonomou**
As per the discussion in #1058, TI's Flash Programmer seems to dislike 03 records in ihex files. As a result, the `.hex` file generated by the CC26xx build system cannot be used with Flash Programmer.

This pull changes the CC26xx build system as follows:
* We `objcopy` from the `.elf` to an intermediate `.hex` ~~which has all addresses offset by `0x200000`. This forces `objcopy` to only use 04/05 records instead of 02/03.~~
* We use `srecord` to post-process this intermediate hex. This produces a final hex file that uses 32-bit linear addresses ~~by offsetting everything back by `-0x200000`. This produces a final `.hex` which has correct addresses, but which only uses 04/05 records.~~

This final `.hex` was used successfully to program Srf+CC2650 EM as well as the SensorTag with cc26xx-demo as well as with cc26xx-web-demo.

Closes #1058

### CODE REVIEW

**Comment by bthebaudeau** [comment]
Why do you use `0x200000` instead of `0x100000`, which is the minimal offset doing the trick? We don't care about the start of flash on CC2538 here. But anyway, you don't need to use any offset here because `srec_cat` 32-bit linear addresses by default (`--address-length=4`: `i32hex`).

You could use the stem automatic variable, i.e. `$*-offset.hex` instead of `$(basename $@)-offset.hex`, twice.

So to sum up, your `.hex` make rule could be simplified like this:
```
%.hex: %.elf
	$(OBJCOPY) -O ihex $< $*-i16hex.hex
	$(SREC_CAT) $*-i16hex.hex -intel -o $@ -intel
```

**Comment by g-oikonomou** [comment]
Yep, you are right, that works too. Updated.

**Comment by bthebaudeau** [comment]
:+1:

**Comment by jonnteolsson** [comment]
Very nice, I will take an action at TI internally to make sure we improve the Flash programmer.

**Comment by bthebaudeau** [comment]
Thanks @jonnteolsson. If I may ask, unrelated to this one, but related to Contiki, maybe you can have some influence internally for this one too:
http://e2e.ti.com/support/wireless_connectivity/f/155/p/308156/1105746#1105746
I never got any answer (and I no longer need one), but for me this is a silicon erratum, which of course should be documented for the TI community.

**Comment by vsaw** [comment]
:+1: Another handy feature of this is that the BSL also uses the iHex format in case anyone tries to flash Contiki using the BSL.

**Comment by jonnteolsson** [comment]
@bthebaudeau I have pinged the right resources regarding your issue.

**Comment by bthebaudeau** [comment]
@jonnteolsson Thanks!

**Comment by g-oikonomou** [comment]
There is also this alternative:

```
%.i16hex: %.elf
       $(OBJCOPY) -O ihex $< $@

%.hex: %.i16hex
       $(SREC_CAT) $< -intel -o $@ -intel
```

Two good things about it:
* Make identifies .i16hex as intermediate and deletes it after the build, not leaving junk laying around.
* Users needing to use this file for whatever reason can make it explicitly by running `make foo-example.i16hex`

**Comment by bthebaudeau** [comment]
Yes, that'd be good, and that would also simplify the `$*-i16hex.hex` and avoid repeating it twice.

**Comment by g-oikonomou** [comment]
Pushed. Also a commit to install srecord when running aapcs tests on travis

**Comment by bthebaudeau** [comment]
:+1:

**Comment by bthebaudeau** [comment]
:+1:

Nothing more to change? Ready to merge?

**Comment by g-oikonomou** [comment]
I made a change to travis.yml which triggered a build. If/when it passes then yes!
